# Generated by using Rcpp::compileAttributes() -> do not edit by hand
# Generator token: 10BE3573-1514-4C36-9D1C-5A225CD40393

dna_blast <- function(query_table, db_table, output_file, maxAccepts = 1L, maxRejects = 16L, minIdentity = 0.75, strand = "both") {
    invisible(.Call('_blaster_dna_blast', PACKAGE = 'blaster', query_table, db_table, output_file, maxAccepts, maxRejects, minIdentity, strand))
}

protein_blast <- function(query_table, db_table, output_file, maxAccepts = 1L, maxRejects = 16L, minIdentity = 0.75) {
    invisible(.Call('_blaster_protein_blast', PACKAGE = 'blaster', query_table, db_table, output_file, maxAccepts, maxRejects, minIdentity))
}

#' Read the contents of a Fasta file into a DataFrame
#' 
#' @param filename A string; name of the imported Fasta file.
#' @param filter A string; only include those entries which contain a sequence motif specified in this argument.
#' @return A dataframe with Id and Seq columns
#' @examples
#' 
#' query <- system.file("extdata", "query.fasta", package = "blaster")
#' db <- system.file("extdata", "db.fasta", package = "blaster")
#' 
#' query <- read_fasta(filename = query)
#' 
#' query <- read_fasta(filename = db, filter = "TGGTTGAGG")
#' 
#' @export
read_fasta <- function(filename, filter = "", non_nucleotide_chars = "error") {
    .Call('_blaster_read_fasta', PACKAGE = 'blaster', filename, filter, non_nucleotide_chars)
}

